tests/DecisionServiceTests/DecisionServiceTest.php

```diff
@@ -603,6 +614,445 @@ public function testGetVariationWithBucketingId()
+        $expected_variation = $experiment->getVariations()[0];   
+        // Provide attributes such that user qualifies for audience
+        $user_attributes = ["browser_type" => "chrome"];
+
+        $bucketer = new \ReflectionProperty(DecisionService::class, '_bucketer');
+        $bucketer->setAccessible(true);
+        $bucketer->setValue($this->decisionService, $this->bucketerMock);
+
+        $this->bucketerMock
+            ->method('bucket')
+            ->willReturn($expected_variation);
+
+        $this->loggerMock->expects($this->at(0))
+            ->method('log')
+            ->with(Logger::DEBUG, 
+            "Attempting to bucket user 'user_1' into rollout rule '{$experiment->getKey()}'.");
+
+        $this->assertEquals(
+            $this->decisionService->getVariationForFeatureRollout($feature_flag, 'user_1', $user_attributes),
+            $expected_variation
+        );
+    }
+
+    // should return the variation the user is bucketed into when the user is bucketed into the "Everyone Else" rule'
+    // and the user is not bucketed into the targeting rule
+    public function testGetVariationForFeatureRolloutWhenUserIsNotBucketedInTheTargetingRuleButBucketedToEveryoneElseRule(){
+        $feature_flag = $this->config->getFeatureFlagFromKey('boolean_single_variable_feature');
+        $rollout_id = $feature_flag->getRolloutId();
+        $rollout = $this->config->getRolloutFromId($rollout_id);
+        $experiment0 = $rollout->getExperiments()[0];
+        // Everyone Else Rule
+        $experiment2 = $rollout->getExperiments()[2];
+        $expected_variation = $experiment2->getVariations()[0];   
+
+        // Provide attributes such that user qualifies for audience
+        $user_attributes = ["browser_type" => "chrome"];
+        $this->decisionService = new DecisionService($this->loggerMock, $this->config);
+        $bucketer = new \ReflectionProperty(DecisionService::class, '_bucketer');
+        $bucketer->setAccessible(true);
+        $bucketer->setValue($this->decisionService, $this->bucketerMock);
+        // Make bucket return null when called for first targeting rule
+        $this->bucketerMock->expects($this->at(0))
+            ->method('bucket')
+            ->willReturn(null);
+        //Make bucket return expected variation when called second time for everyone else
+        $this->bucketerMock->expects($this->at(1))
+            ->method('bucket')
+            ->willReturn($expected_variation);
+
+        $this->loggerMock->expects($this->at(0))
+            ->method('log')
+            ->with(Logger::DEBUG, 
+            "Attempting to bucket user 'user_1' into rollout rule '{$experiment0->getKey()}'.");
+        $this->loggerMock->expects($this->at(1))
+            ->method('log')
+            ->with(Logger::DEBUG, 
+            "User 'user_1' was excluded due to traffic allocation. Checking 'Eveyrone Else' rule now.");
+
+
+        $this->assertEquals(
```

Comment:
I would strongly recommend to log message here. 
"User 'user_1' meets conditions for targeting rule 'Everyone Else'." And add in the everoneelse rule condition.

Reply:
Will do